### PR TITLE
fix(ci): MME RHEL8 build from scratch was broken

### PR DESCRIPTION
**Pull request by rdefosse**
## Summary

After merging [the RHEL8 optimization pull request](https://github.com/magma/magma/pull/14070), I verified with manual post-merge check: [It](https://jenkins-oai.eurecom.fr/job/MAGMA-MME-production/19074/) failed.

The reason is simple: the `build-variant` (ie `FEATURES`) was an `argument` and not `environment variables`.

With the new multi-staged build, it was not carried over.

But since these dockerfiles only works for the `mme-oai` build-variant, it is better to use `ENV` anyway.

I did also some clean-up on the build arguments. `GIT_PROXY` is no longer used in Eurecom/OAI environment. It's useless.

## Test Plan

This time, I did build from scratch. Working now.

### CODE REVIEW

**Review by crasu — APPROVED** [review]

File: lte/gateway/docker/mme/Dockerfile.ubuntu18.04 [hunk 1]
```diff
@@ -51,8 +50,6 @@ RUN echo "Install 3rd party dependencies" && \
     apt-get install -y libtins-dev && \
     ln -s /usr/bin/python2.7 /usr/local/bin/python
 
-RUN ["/bin/bash", "-c", "if [[ -v GIT_PROXY ]]; then git config --global http.proxy $GIT_PROXY; fi"]
-
```
Comment (Neudrino) [inline lte/gateway/docker/mme/Dockerfile.ubuntu18.04:55]: This proxy change seems to be unrelated to the intend of the PR and was not mentioned at all.
Is this intentional? If so, could a comment be made in the PR description?

Comment (rdefosse) [reply lte/gateway/docker/mme/Dockerfile.ubuntu18.04:55]: @Neudrino 

yes it was intentional.

I edited the description of the PR.

**Review by Neudrino — APPROVED** [review]
